nsxt/resource_nsxt_policy_tier0_inter_vrf_routing.go

```diff
@@ -0,0 +1,391 @@
+		brlMap := make(map[string]interface{})
+		brlMap["address_family"] = brl.AddressFamily
+		brlMap["in_filter"] = stringList2Interface(brl.InFilter)
+		brlMap["out_filter"] = stringList2Interface(brl.OutFilter)
+
+		brlList = append(brlList, brlMap)
+	}
+	d.Set("bgp_route_leaking", brlList)
+
+	var sraList []interface{}
+	if obj.StaticRouteAdvertisement != nil {
+		sra := make(map[string]interface{})
+		var advRuleList []interface{}
+		for _, advRule := range obj.StaticRouteAdvertisement.AdvertisementRules {
+			elem := make(map[string]interface{})
+			elem["action"] = advRule.Action
+			elem["name"] = advRule.Name
+			elem["prefix_operator"] = advRule.PrefixOperator
+			elem["route_advertisement_types"] = stringList2Interface(advRule.RouteAdvertisementTypes)
+			elem["subnets"] = stringList2Interface(advRule.Subnets)
+
+			advRuleList = append(advRuleList, elem)
+		}
+		sra["advertisement_rule"] = advRuleList
+		sra["in_filter_prefix_list"] = stringList2Interface(obj.StaticRouteAdvertisement.InFilterPrefixList)
+
+		sraList = []interface{}{sra}
+	}
+	d.Set("static_route_advertisement", sraList)
+	d.Set("target_path", obj.TargetPath)
+
+	return nil
+}
+
+func resourceNsxtPolicyTier0InterVRFRoutingUpdate(d *schema.ResourceData, m interface{}) error {
+	connector := getPolicyConnector(m)
+	client := tier_0s.NewInterVrfRoutingClient(connector)
+
+	gwPolicyPath := d.Get("gateway_path").(string)
+	isT0, gwID := parseGatewayPolicyPath(gwPolicyPath)
+	if gwID == "" {
+		return fmt.Errorf("gateway_path is not valid")
+	}
+	if !isT0 {
+		return fmt.Errorf("Tier0 gateway path expected, got %s", gwPolicyPath)
+	}
+
+	id := d.Get("nsx_id").(string)
+
+	obj := getPolicyInterVRFRoutingFromSchema(d)
+	revision := int64(d.Get("revision").(int))
+	obj.Revision = &revision
+
+	err := client.Patch(gwID, id, obj)
+	if err != nil {
+		return handleUpdateError("InterVRFRouting", *obj.DisplayName, err)
+	}
+
+	return resourceNsxtPolicyTier0InterVRFRoutingRead(d, m)
+}
+
+func resourceNsxtPolicyTier0InterVRFRoutingDelete(d *schema.ResourceData, m interface{}) error {
+	connector := getPolicyConnector(m)
+	client := tier_0s.NewInterVrfRoutingClient(connector)
+
+	gwPolicyPath := d.Get("gateway_path").(string)
+	isT0, gwID := parseGatewayPolicyPath(gwPolicyPath)
+	if gwID == "" {
+		return fmt.Errorf("gateway_path is not valid")
+	}
+	if !isT0 {
+		return fmt.Errorf("Tier0 gateway path expected, got %s", gwPolicyPath)
+	}
+
+	id := d.Get("nsx_id").(string)
+
+	err := client.Delete(gwID, id)
+	if err != nil {
+		return handleDeleteError("InterVRFRouting", id, err)
+	}
+
+	return nil
+}
+
+func resourceNsxtPolicyTier0InterVRFRoutingImport(d *schema.ResourceData, m interface{}) ([]*schema.ResourceData, error) {
+	importID := d.Id()
+
+	rd, err := nsxtPolicyPathResourceImporterHelper(d, m)
+	if err == nil {
+		d.Set("nsx_id", d.Id())
+		gwPath, err := getParameterFromPolicyPath("", "/inter-vrf-routing/", importID)
+		if err != nil {
+			return nil, err
```

Comment:
I think this error message should be something like `incorrect policy path for inter vrf routing resource: %v`